app/src/main/kotlin/com/mapbox/vision/teaser/view/PlaybackSeekBarView.kt

```diff
@@ -0,0 +1,38 @@
+package com.mapbox.vision.teaser.view
+
+import android.content.Context
+import android.text.format.DateUtils
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.LinearLayout
+import android.widget.SeekBar
+import com.mapbox.vision.teaser.R
+import kotlinx.android.synthetic.main.view_playback_seek_bar.view.*
+
+class PlaybackSeekBarView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0,
+    defStyleRes: Int = 0
+) : LinearLayout(context, attrs, defStyle, defStyleRes) {
+
+    init {
+        LayoutInflater.from(context).inflate(R.layout.view_playback_seek_bar, this, true)
+    }
+
+    var onSeekBarChangeListener: SeekBar.OnSeekBarChangeListener? = null
+        get() = field
+        set(value) {
+            seek_bar.setOnSeekBarChangeListener(value)
+        }
+
+    fun setTimePosition(seconds: Float) {
+        playback_position_time_text.text = DateUtils.formatElapsedTime(seconds.toLong())
+        seek_bar.progress = seconds.toInt()
+    }
+
+    fun setDuration(seconds: Float) {
```

Comment:
why do you pass it as Float and then convert to Long and Int?

Reply:
It is float outside, but probably a bit consistent to use float here

Reply:
hm, inside it makes two conversions. Probably it makes code a bit clear because setDuration\setProgress are used in MainActivity and ArReplayNavigationActivity